packages/commercetools/theme/components/Checkout/UserBillingAddresses.vue

```diff
@@ -42,32 +42,34 @@ import {
   SfCheckbox,
   SfAddressPicker
 } from '@storefront-ui/vue';
-import { userBillingGetters } from '@vue-storefront/commercetools';
-
+import { useUserBilling, userBillingGetters } from '@vue-storefront/commercetools';
 export default {
   name: 'UserBillingAddresses',
   props: {
     currentAddressId: {
-      type: Number,
+      type: String | Number,
       required: true
     },
     setAsDefault: {
       type: Boolean,
       required: true
-    },
-    billingAddresses: {
-      type: Array,
-      required: true
     }
   },
   components: {
     SfCheckbox,
     SfAddressPicker
   },
   setup (_, { emit }) {
-    const setCurrentAddress = $event => emit('setCurrentAddress', $event);
-
+    const { billing: userBilling } = useUserBilling();
+    const setCurrentAddress = async addressId => {
+      const selectedAddress = userBillingGetters.getAddresses(userBilling.value, { id: addressId });
+      if (!selectedAddress || !selectedAddress.length) {
+        return;
+      }
+      emit('setCurrentAddress', selectedAddress[0]);
```

Comment:
why you have `update:setAsDefault` and just `setCurrentAddress` ? if you decide to use a convention use it consequently

Reply:
`update:setAsDefault` is used because I am using `.sync` modifier. For second one I do not use it (I send you details earlier in Slack's DM)